### DISCUSSION TITLE
Error using sparse

### DISCUSSION THREAD

**Discussion by AleksandrPupynin** [mtex-toolbox/mtex#1671] [Ask Anything] [answered]
Error: Repeated indices are not supported for sparse logical matrices.
Stage: grain reconstruction using calcGrains
mtex version 5.8.2
MatLab version 2018b

Using mtex version 5.8.1 calcGrains finished processing normally.

Can anyone help to resolve this problem?

**Comment by ralfHielscher** [comment]
Could you please provide some more information.
E.g. some line number for the error message. Does the error occur for the data included into MTEX, e.g.
```matlab
mtexdata twins
calcGrains(ebsd)
```
Ralf.

**Comment by AleksandrPupynin** [comment, marked as answer]
Dear Ralf!

The code and description of the error that occurred are attached in the
file.


пт, 2 дек. 2022 г. в 15:48, Ralf Hielscher ***@***.***>:

> Could you please provide some more information.
> E.g. some line number for the error message. Does the error occur for the
> data included into MTEX, e.g.
>
> mtexdata twinscalcGrains(ebsd)
>
> Ralf.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mtex-toolbox/mtex/discussions/1671#discussioncomment-4292188>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4PUHKY2IF6CLAQSXADWZXDWLHVYHANCNFSM6AAAAAASPMS6DI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

With best regards, Pupynin A.

**Reply by ralfHielscher** [reply]
There is no file attached :)

**Reply by AleksandrPupynin** [reply]
Excuse me, please... :(
[sparce_err_desc.docx](https://github.com/mtex-toolbox/mtex/files/10141448/sparce_err_desc.docx)

**Reply by ralfHielscher** [reply]
Could you please replace in EBSDAnalysis/@grainBoundary/grainBoundary (line 192)

I_FG = sparse(iF(ind),gB.grainId(ind),true);

with

I_FG = logical(sparse(iF(ind),gB.grainId(ind),1));

Ralf.

**Reply by AleksandrPupynin** [reply]
In mtex folder?

**Reply by ralfHielscher** [reply]
sure :)

**Reply by AleksandrPupynin** [reply]
What we get:
Grains reconstruction in progress
Grains reconstruction is finished

Without errors... :)

**Reply by AleksandrPupynin** [reply]
Thanks a lot!
